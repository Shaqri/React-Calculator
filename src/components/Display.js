import PropTypes from 'prop-types';
import React from 'react';

function Display(props) {
  const { result } = props;

  return (
    <div className="display">
      {result}
    </div>
  );
}

Display.defaultProps = {
  result: '0',
};

Display.propTypes = {
  result: PropTypes.string,
};

export default Display;
