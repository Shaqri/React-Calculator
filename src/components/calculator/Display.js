import PropTypes from 'prop-types';
import React from 'react';
import * as styles from '../../styles/components.module.css';

function Display(props) {
  const { result } = props;

  return (
    <div className={styles.display}>
      {result}
    </div>
  );
}

Display.defaultProps = {
  result: '0',
};

Display.propTypes = {
  result: PropTypes.string,
};

export default Display;
