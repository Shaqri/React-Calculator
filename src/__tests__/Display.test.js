import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';
import Display from '../components/Display';

describe('Display', () => {
  it('render result passed in props', async () => {
    const result = '15';
    render(<Display result={result} />);
    const display = await screen.getByText(result);
    expect(display).toBeInTheDocument();
    expect(display).toMatchSnapshot();
  });
});
