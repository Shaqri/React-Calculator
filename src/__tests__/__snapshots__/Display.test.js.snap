// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display render result passed in props 1`] = `
<div>
  15
</div>
`;
